lib/openhub_security.rb

```diff
@@ -1,26 +1,47 @@
 # frozen_string_literal: true
 
 module OpenhubSecurity
+  def self.validate_response(json)
+    json.dig('response', 'numFound') && json['response']['numFound'].to_i.positive?
+  end
+
   def self.get_uuid(project_name)
     url = ENV['KB_PROJECT_SEARCH_URL'] + "&q=#{CGI.escape(project_name)}"
     json = get_response(url)
-    return json['response']['docs'][0]['uuid'] if json['response']['numFound'].to_i.positive?
+    return json['response']['docs'][0]['uuid'] if validate_response(json)
   end
 
-  def self.get_response(url)
+  def self.get_http(url)
     # Parse the URL to get the host and port and convert to a Net::HTTP object
     # With that object, we can set "use_ssl" explicitly
     parsed_url = URI.parse(url)
     http = Net::HTTP.new(parsed_url.host, parsed_url.port)
     http.use_ssl = true
+    http
+  end
 
+  def self.get_http_request(url)
     # Create a GET request so we can add the auth token to the header
     # We could combine lines:
     # "JSON.parse(http.request(request).read_body)", but this is easer to debug
+    parsed_url = URI.parse(url)
     request = Net::HTTP::Get.new(parsed_url.request_uri)
     request['X-BDS-AuthToken'] = ENV['KB_AUTH_KEY']
     request['User-Agent'] = "ohloh-ui/#{ENV['COMMIT_SHA']}"
-    response = http.request(request)
-    JSON.parse(response.read_body)
+    request
+  end
+
+  def self.get_response(url)
+    response = {}
+    http = get_http(url)
+
+    request = get_http_request(url)
+    begin
+      response = http.request(request)
+      response = JSON.parse(response.read_body)
+    rescue SocketError => e
+      Rails.logger.error("OpenhubSecurity SocketError Exception:  #{e.message}")
+    end
+    response
```

Comment:
Does response need a default value? It seems to gets overriden on assignment in every scenario.
  The above function can be simplied.
  ```
  def self.get_response(url)
    http = get_http(url)

    request = get_http_request(url)
    response = http.request(request)
    response = JSON.parse(response.read_body)
  rescue SocketError => e
    Rails.logger.error("OpenhubSecurity SocketError Exception:  #{e.message}")
    response
  end
  ```

Reply:
If the intention was to catch only the SocketError during `http.request`, then let's extract it into another method.

```ruby
  def self.get_response(url)
    http = get_http(url)

    request = get_http_request(url)
    get_safe_response(http, request)
  end

  def self.get_safe_response(http, request)
    response = http.request(request)
    response = JSON.parse(response.read_body)
  rescue SocketError => e
    Rails.logger.error("OpenhubSecurity SocketError Exception:  #{e.message}")
    response || {}
  end
```

Reply:
Thanks!